desktop/apps/artwork/components/current_auctions/current_auctions.jsx

```diff
@@ -0,0 +1,83 @@
+import React, { PropTypes } from 'react'
+import { partition, take } from 'underscore'
+import { upcomingLabel } from '../../lib/auction_label'
+
+export default function CurrentAuctions ({ auctionContextId, sales }) {
+  const sortedSales = sortSales(sales, auctionContextId)
+
+  return (
+    <section className='artwork-current-auctions'>
+      <div className='artwork-current-auctions__header-area'>
+        <h3>
+          Current Auctions on Artsy
+        </h3>
+
+        <a href='/auctions'>
+          View All Auctions
+        </a>
+      </div>
+
+      <div className='artwork-current-auctions__sales'>
+        {sortedSales.map((sale) => {
+          const {
+            cover_image: {
+              cropped: {
+                url
+              }
+            },
+            end_at,
+            href,
+            id,
+            name,
+            live_start_at,
+            start_at,
+            status
+          } = sale
+
+          const statusLabel = 'Auction ' + upcomingLabel(start_at, end_at,
+                                                         live_start_at, status)
+          return (
+            <div className='artwork-current-auctions__sale-item' key={id}>
+              <a href={href}>
+                <img src={url} alt={name} />
+
+                <div className='artwork-current-auctions__label-container'>
+                  <div className='artwork-current-auctions__name'>
+                    {name}
+                  </div>
+
+                  <div className='artwork-current-auctions__status'>
+                    {statusLabel}
+                  </div>
+                </div>
+              </a>
+            </div>
+          )
+        })}
+      </div>
+    </section>
+  )
+}
+
+CurrentAuctions.propTypes = {
+  /**
+   * If the module is displayed in an Auction context (such as viewing an
+   * artwork that is in a sale) and an auction id is provided, ensure the first
+   * displayed item is the same as the context.
+   */
+  auctionContextId: PropTypes.string,
```

Comment:
For unclear propTypes this area can serve as a place to document whats going on. 

When using ES6 classes `propTypes` make a little more sense as they're more declarative, being at the top rather than the bottom: 

```javascript
class Thing extends Component {
  static propTypes = {
     name: PropTypes.string.isRequired,
     otherThing: PropTypes.shape({
         foo: ProTypes.string.isRequired,
         bar: ProTypes.string.isRequired,
         baz: ProTypes.oneOfType([
            PropTypes.func,
            PropTypes.bool
         ])
     })
   }
   ... 
}
```

Having propType checks has helped me catch innumerable problems in the past, especially as things scale out.